### DISCUSSION TITLE
Making a lot of Account/Profile Settings togglable

### DISCUSSION THREAD

**Discussion by MotikaCraft-TheJoshua** [NamelessMC/Nameless#2292] [Ideas]
I don't know what do you think but for me there are a lot "bloatware" settings that aren't EVER used if you have your site in your region.

For my example, and i am sure there are similar examples, I am from the region where 4+ countries around mine are talking 99% the same language (extremely similar) and people from here who are goind to be using Nameless won't ever use, and don't need this account setting - CHOOSING TIMEZONE.

Also, i once recommended to add this option to disable timezone but it's labeled as out-of-scope, while we still today can't set default timezone (#1731); i tried setting default timezone and whenever someone registers, their default timezone is ALWAYS & ALWAYS "London". This is unchangeable and this is why we need FLEXIBILITY to toggle everything on/off, even if it's important just as selecting your timezone (which is not the same for all users, I know).

I also once mentioned that making CHOOSING LANGUAGE toggleable (#1724) and i can also said something about that if needed but i think it's enough to realize that we need it.

"Use Gravatar as avatar" option, in my opinion is not something that need to be there, if player want their skin head, he'll basically find it on internet and upload it just as any other image - i think this should also be toggleable

And believe me, this section ("Account/Profile Settings") will still be rich even if almost all of the fields are disabled except "Get emails for topics you follow" and **Signature** (even if the option for removing "Change Passoword" and "Change Email" from #2251 gets implemented). fields. There are yet more to come!

**Comment by samerton** [comment]
#2241 will allow timezones to be automatically detected during registration.

Instead of simply adding toggles for each of these settings, the whole code side of profile settings could do with a rework to allow modules to integrate easily. Maybe as part of this, we could add ways to allow admins to choose which ones are shown.

Then, the problem is, you could end up with a massive list of toggles in the StaffCP which moves the management headache to there instead.

**Comment by MotikaCraft-TheJoshua** [comment]
> #2241 will allow timezones to be automatically detected during registration.
> 
> Instead of simply adding toggles for each of these settings, the whole code side of profile settings could do with a rework to allow modules to integrate easily. Maybe as part of this, we could add ways to allow admins to choose which ones are shown.
> 
> Then, the problem is, you could end up with a massive list of toggles in the StaffCP which moves the management headache to there instead.

Then should we rename this into something like "Rework "Profile Settings" for easier module integration + make it more customizable"?
